src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -149,7 +149,10 @@ public function normalize($object, string $format = null, array $context = [])
             }
         }
 
-        if ($this->isCircularReference($object, $context)) {
+        if ((isset($this->context['lastCircularObjectFound']) && $this->context['lastCircularObjectFound'] == \get_class($object))
```

Comment:
strict comparison
`$this->context['lastCircularObjectFound'] === \get_class($object)`